tests/opt-testcases/sample.sh

```diff
@@ -49,6 +62,22 @@ run_test    "Sample: ssl_client1, gnutls server, TLS 1.3" \
             -S "Error" \
             -C "error"
 
+# The server complains of extra data after it closes the connection
+# because the client keeps sending data, so the server receives
+# more application data when it expects a new handshake. We consider
+# the test a success if both sides have sent and received application
+# data, no matter what happens afterwards.
+run_test    "Sample: dtls_client, ssl_server2" \
+            -P 4433 \
+            "$PROGRAMS_DIR/ssl_server2 dtls=1 server_addr=localhost" \
```

Comment:
Out of curiosity: why do we need `server_addr=localhost` here?

Reply:
By default, on my machine, `ssl_server2` only listens on IPv4. I actually don't understand why: I would have thought it would prioritize IPv6. `dtls_client` connects to `localhost`, whatever it is. I remember it took me a little while to find something that would minimize changes in the test programs, have the tests pass on my machine (where `localhost` resolves to `::1`), and have the tests pass on the CI (where IPv6 is disabled).

I don't know why this isn't necessary for `ssl_client1`.